kong/plugins/basic-auth/api.lua

```diff
@@ -1,9 +1,41 @@
 local crud = require "kong.api.crud_helpers"
+local basic_auth_utils = require "kong.plugins.basic-auth.utils"
+
+local function get_config(dao_factory)
+  -- in case of no conf.
+  local default_config = {
+    hide_credentials = false,
+    encryption_method = "plain",
+  }
+
+  local data, err = dao_factory.plugins:find_by_keys({ name = "basic-auth" })
```

Comment:
I am just realising now but... This is retrieving _any_ configuration? That won't do it. We need to update the core to introduce some kind of link between a plugin configuration and related entities (such as credentials) created by plugins.